tee-launcher/launcher.py

```diff
@@ -1,21 +1,107 @@
 from collections import deque
 import logging
 import os
+from typing import Dict, Union
 import requests
-from subprocess import run
+from subprocess import CompletedProcess, run
 import sys
 import time
 import traceback
+from dataclasses import dataclass
+
+from requests.models import Response
 
 # The volume where this file resides is shared between launcher and app.
 # To avoid concurrent modifications, the launcher mounts the volume read-only!
-IMAGE_DIGEST_FILE="/mnt/shared/image-digest"
+# the contents of this file are generated by the node itself and fetched from the contract.
+IMAGE_DIGEST_FILE = "/mnt/shared/image-digest"
+
+# only considered if `IMAGE_DIGEST_FILE` does not exist.
+ENV_VAR_DEFAULT_IMAGE_DIGEST = "DEFAULT_IMAGE_DIGEST"
+# the time to wait between rpc requests, in milliseconds. Defaults to 500 milliseconds.
+OS_ENV_VAR_RPC_REQUEST_INTERVAL_MS = 'RPC_REQUST_INTERVAL_MS'
+# the maximum time to wait for an rpc response. Defaults to 10 seconds.
+OS_ENV_VAR_RPC_REQUST_TIMEOUT_SECS = 'RPC_REQUST_TIMEOUT_SECS'
+# the maximum number of attempts for rpc requests until we raise an exception
+OS_ENV_VAR_RPC_MAX_ATTEMPTS = 'RPC_MAX_RETRIES'
+# MUST be set to 1.
+OS_ENV_DOCKER_CONTENT_TRUST = 'DOCKER_CONTENT_TRUST'
+
+# Dstack user configuration flags
+DSTACK_USER_CONFIG_FILE = '/tapp/user_config'
+
+# Dstack user config. Read from `DSTACK_USER_CONFIG_FILE`
+USER_ENV_VAR_LAUNCHER_IMAGE_TAGS = 'LAUNCHER_IMAGE_TAGS'
+USER_ENV_VAR_LAUNCHER_IMAGE_NAME = 'LAUNCHER_IMAGE_NAME'
+USER_ENV_VAR_LAUNCHER_IMAGE_REGISTRY = 'LAUNCHER_REGISTRY'
```

Comment:
I'm confused. The comment says we should read these from a file but we still have env variables for it? Why not just take a file path?

Reply:
Good point. I renamed to `DSTACK_USER_CONFIG_*`